TextFieldEffects/TextFieldEffects/LTRKaedeTextField.swift

```diff
@@ -0,0 +1,229 @@
+//////////////////////////////////////////////////////////////////////
+
+class LTRKaedeTextField: KaedeTextField {
+    
+    /*
+     // Only override drawRect: if you perform custom drawing.
+     // An empty implementation adversely affects performance during animation.
+     override func drawRect(rect: CGRect) {
+     // Drawing code
+     }
+     */
+    
+    
+    
+    override  func animateViewsForTextEntry() {
+        UIView.animate(withDuration: 0.35, delay: 0.0, usingSpringWithDamping: 0.8, initialSpringVelocity: 2.0, options: .beginFromCurrentState, animations: ({
+            
+            self.placeholderLabel.frame.size.width = self.frame.size.width * 0.25
+            
+        }), completion: nil)
+        
+        UIView.animateWithDuration(0.45, delay: 0.0, usingSpringWithDamping: 0.8, initialSpringVelocity: 1.5, options: .BeginFromCurrentState, animations: ({
+            
+            self.foregroundView.frame.size.width = self.frame.size.width * 0.3
+            
+        }), completion: { _ in
+            self.animationCompletionHandler?(type: .TextEntry)
+        })
+    }
+    
+    override  func animateViewsForTextDisplay() {
+        if text!.isEmpty {
+            UIView.animate(withDuration: 0.3, delay: 0.0, usingSpringWithDamping: 0.8, initialSpringVelocity: 2.0, options: .beginFromCurrentState, animations: ({
+                self.placeholderLabel.frame.size.width = self.frame.size.width
+                //                self.placeholderLabel.frame.origin = self.placeholderInsets
+            }), completion: nil)
+            
+            UIView.animateWithDuration(0.3, delay: 0.0, usingSpringWithDamping: 1.0, initialSpringVelocity: 2.0, options: .BeginFromCurrentState, animations: ({
+                self.getForgroundView().frame.size.width=self.frame.size.width
+                //                self.foregroundView.frame.origin = CGPointZero
+            }), completion: { _ in
+                
+                self.animationCompletionHandler?(type: .TextDisplay)
+            })
+        }
+    }
+    
+    override  func editingRect(forBounds bounds: CGRect) -> CGRect {
+        var b = bounds
+        b.origin.x += self.placeholderLabel.frame.size.width + 10
+        
+        let frame = CGRect(origin: b.origin, size: CGSize(width: bounds.size.width * 0.6, height: bounds.size.height))
+        
+        
+//        self.alignmentRectInsets.left
+        let start = self.getTextFieldInsets().x
+        return CGRectInset(frame, start , self.getTextFieldInsets().y)
+    }
+    
+    override  func textRect(forBounds bounds: CGRect) -> CGRect {
+        var b = bounds
+        b.origin.x += self.placeholderLabel.frame.size.width + 10
+        
+        let frame = CGRect(origin: b.origin, size: CGSize(width: bounds.size.width * 0.6, height: bounds.size.height))
+        
+        let start = self.getTextFieldInsets().x
+        return CGRectInset(frame, start, self.getTextFieldInsets().y)
+    }
+    
+}
```

Comment:
Hmmmm what is all this? :D Why create a new file? and not just fix Kaede to have right left support